tests/test_awsdatabricks.py

```diff
@@ -0,0 +1,597 @@
+        }
+    )
+
+    ec2 = boto.client("ec2", region_name=MOCK_DBX_CONF.aws_region_name)
+    ec2_stubber = Stubber(ec2)
+    ec2_stubber.add_response("describe_instances", {"Reservations": []})
+
+    def client_patch(name, **kwargs):
+        match name:
+            case "ec2":
+                return ec2
+
+    with ec2_stubber, patch("boto3.client") as mock_aws_client:
+        mock_aws_client.side_effect = client_patch
+        result = create_prediction_for_run("75778", "Premium", "Jobs Compute", "my-project-id")
+
+    assert result.error
+
+
+@patch("sync.clients.databricks.get_databricks_config")
+@patch("sync.clients.databricks.get_default_client")
+@patch("sync.awsdatabricks.create_prediction")
+def test_create_prediction_for_run_success(
+    create_prediction, get_default_client, get_databricks_config
+):
+    create_prediction.return_value = Response(result=uuid4())
+
+    get_databricks_config.return_value = MOCK_DBX_CONF
+
+    get_default_client.return_value = MockDatabricksClient(
+        {
+            "get_run": {"75778": MOCK_RUN},
+            "get_cluster": {"0101-214342-tpi6qdp2": MOCK_CLUSTER},
+            "get_cluster_events": {"0101-214342-tpi6qdp2": {"events": [], "total_count": 0}},
+        }
+    )
+
+    ec2 = boto.client("ec2", region_name=MOCK_DBX_CONF.aws_region_name)
+    ec2_stubber = Stubber(ec2)
+    ec2_stubber.add_response("describe_instances", MOCK_INSTANCES)
+
+    s3_file_prefix = "path/to/logs/0101-214342-tpi6qdp2/eventlog/0101-214342-tpi6qdp2"
+
+    s3 = boto.client("s3")
+    s3_stubber = Stubber(s3)
+    s3_stubber.add_response(
+        "list_objects_v2",
+        {
+            "Contents": [
+                {
+                    "Key": f"{s3_file_prefix}/eventlog",
+                    "LastModified": datetime.utcfromtimestamp(1681249791560 / 1000),
+                }
+            ]
+        },
+        {"Bucket": "bucket", "Prefix": s3_file_prefix},
+    )
+    s3_stubber.add_response(
+        "get_object",
+        {
+            "ContentType": "application/octet-stream",
+            "ContentLength": 0,
+            "Body": StreamingBody(io.BytesIO(), 0),
+        },
+        {"Bucket": "bucket", "Key": f"{s3_file_prefix}/eventlog"},
+    )
+
+    def client_patch(name, **kwargs):
+        match name:
+            case "s3":
+                return s3
+            case "ec2":
+                return ec2
+
+    with s3_stubber, ec2_stubber, patch("boto3.client") as mock_aws_client:
+        mock_aws_client.side_effect = client_patch
+        result = create_prediction_for_run("75778", "Premium", "Jobs Compute", "my-project-id")
+
+    assert not result.error
+    assert result.result
+
+
+@patch("sync.clients.databricks.get_databricks_config")
+@patch("sync.clients.databricks.get_default_client")
+@patch("sync.awsdatabricks.event_log_poll_duration_seconds")
+@patch("sync.awsdatabricks.create_prediction")
+def test_create_prediction_for_run_event_log_upload_delay(
```

Comment:
This test is intended to exercise the various scenarios in which we want to continue polling the S3 event log location. These conditions include -

- No event log data found at all yet,
- Incomplete event log data where the `eventlog` file (with no suffix) is missing,
- Incomplete event log data where the `LastModified` time of the event log is prior to the Run finishing